### ISSUE TITLE
[Bug]: Statically generated pages from CLI missing HTML-tags

### ISSUE THREAD

**Issue by tmannherz** [pimcore/pimcore#14922]
### Expected behavior

HTML-Tags can be configured on a Pimcore document and those tags are automatically rendered by the `{{ pimcore_head_meta() }}` directive: 

![image](https://user-images.githubusercontent.com/533798/231765431-81ff5f5b-f9f1-4501-9c37-55a9438e9ee8.png)

Viewing page source, we can see that custom tag added to the `head`:

![image](https://user-images.githubusercontent.com/533798/231766196-9f60a6d8-4df6-4685-8ae9-261a00a4dd78.png)

### Actual behavior

If a page is configured for static page generation and the CLI is used to generate pages (`bin/console pimcore:documents:generate-static-pages`), generated pages are missing HTML-Tags.

![image](https://user-images.githubusercontent.com/533798/231767193-3187b038-4a95-4a16-bffb-3099d1035de6.png)

![image](https://user-images.githubusercontent.com/533798/231767296-123d9a6c-326f-4fd8-a334-983bb1f61538.png)

### Steps to reproduce

1. Run the Pimcore demo site. I was able to reproduce on demo branch 10.3 with Pimcore at v10.5.20
2. From the Pimcore admin, edit a document. My example is using the /en/News document.
    - Add an HTML-Tag, for ex. `<meta name="test" />` 
    - Enable static page generation (`Enable server side static rendering`)
3. Access /en/News from your browser. The page should be auto-generated within `public/var/tmp/pages`
4. Back in the browser, view source and notice that your custom tag is present in the source.
5. Re-generate static pages using the console command `bin/console pimcore:documents:generate-static-pages`
6. Refresh /en/News and view source again. Your custom tag is missing.

Likely related, we use https://github.com/dachcom-digital/pimcore-schema to generate json-ld schemas for our pages. That content is also rendered through the `pimcore_head_meta` helper and is missing when the CLI generates static pages.

**Comment by tmannherz** [comment]
This seems to be because document head meta is rendered through [`DocumentMetaDataListener::onKernelRequest`](https://github.com/pimcore/pimcore/blob/10.5/bundles/CoreBundle/EventListener/Frontend/DocumentMetaDataListener.php#L61), and that event isn't dispatched during the static page generation command.

There is a useful event already dispatched by [the static document render](https://github.com/pimcore/pimcore/blob/10.5/lib/Document/Renderer/DocumentRenderer.php#L114) that can be tapped into.

I added this method to `DocumentMetaDataListener` (and a corresponding subscriber entry) and that seems to fix the issue:

```php
    // DocumentMetaDataListener.php
    public function onDocumentPreRender(DocumentEvent $event): void
    {
        if (php_sapi_name() !== 'cli') { // likely only needed in the cli
            return;
        }
        $document = $event->getDocument();
        if ($document instanceof Page && is_array($document->getMetaData())) {
            foreach ($document->getMetaData() as $meta) {
                $this->headMeta->addRaw($meta);
            }
        }
    }
```

**Comment by dvesh3** [comment]
Fixed by #15191
